src/platform/android/java/chip/platform/DiagnosticDataProviderImpl.java

```diff
@@ -0,0 +1,107 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package chip.platform;
+
+import android.content.Context;
+import android.util.Log;
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+
+public class DiagnosticDataProviderImpl implements DiagnosticDataProvider {
+  private static final String TAG = DiagnosticDataProviderImpl.class.getSimpleName();
+  private Context mContext;
+
+  public DiagnosticDataProviderImpl(Context context) {
+    mContext = context;
+  }
+
+  @Override
+  public int getRebootCount() {
+    // TODO: boot count is not supported until api 24, please do it at factory app
+    //        try {
+    //            return Settings.Global.getInt(mContext.getContentResolver(),
+    // Settings.Global.BOOT_COUNT);
+    //        } catch (Settings.SettingNotFoundException e) {
+    //            Log.e(TAG, "getRebootCount: " + e.getMessage());
+    //        }
+    return 100;
+  }
+
+  @Override
+  public NetworkInterface[] getNetworkInterfaces() {
+    try {
+      ArrayList<java.net.NetworkInterface> networkInterfaces =
+          Collections.list(java.net.NetworkInterface.getNetworkInterfaces());
+      int size = networkInterfaces.size();
+      List<NetworkInterface> destInterfaces = new ArrayList<>();
+      for (int i = 0; i < size; i++) {
+        java.net.NetworkInterface nif = networkInterfaces.get(i);
+        String name = nif.getName();
+
+        if (name != null && (name.startsWith("eth") || name.startsWith("wlan"))) {
+          NetworkInterface anInterface = new NetworkInterface();
+          anInterface.isOperational = nif.isUp();
+          anInterface.hardwareAddress = nif.getHardwareAddress();
+          anInterface.name = nif.getName();
+          anInterface.type =
+              name.startsWith("wlan")
+                  ? NetworkInterface.INTERFACE_TYPE_WI_FI
+                  : NetworkInterface.INTERFACE_TYPE_ETHERNET;
+
+          anInterface.offPremiseServicesReachableIPv4 = NetworkInterface.REACHABLE_UNKNOWN;
+          anInterface.offPremiseServicesReachableIPv6 = NetworkInterface.REACHABLE_UNKNOWN;
+          for (Enumeration<InetAddress> inetAddresses = nif.getInetAddresses();
+              inetAddresses.hasMoreElements(); ) {
+            InetAddress inetAddress = inetAddresses.nextElement();
+            if (inetAddress instanceof Inet4Address) {
+              anInterface.offPremiseServicesReachableIPv4 = NetworkInterface.REACHABLE_YES;
```

Comment:
I think it's not sufficient to know if off premise services (if they exist) can be reached by this interface. I think it's better to return false and let the application provide its own implementation if it supports any off premise services.

Reply:
Sorry, maybe I have some misunderstanding for off-premise services, what is the precise definition of off-premise services?

Reply:
I understand that code under src/platform/android/java/ may only be an example and the application can have its own implementation. But I am trying my best to explain in the simple code to show how it cloud be done in app.

Reply:
I don't think there exists a precise definition of the off-premise service in the spec, so I'm assuming it can be any off-premise service (like a cloud service) that a given application needs to connect to. I know an application can override this, but if an application uses any off-premise service then it will need to provide its own implementation of this class anyway, and if it doesn't, then it should set both the attributes to false. Therefore, I think the default implementation will be more useful if it sets those attributes to `false` (then it will be sufficient for at least some applications).

Reply:
> and if it doesn't, then it should set both the attributes to false

Reply:
Right, forgot it's nullable.

Reply:
fixed, thanks

Reply:
While, one question, how could the app access off-premise service? We do not give the device app any chance to let it agree the User Policy and Agreement?